core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -223,7 +223,18 @@ default Optional<ConnectorPartitioningHandle> getCommonPartitioningHandle(Connec
      *
      * @throws RuntimeException if table handle is no longer valid
      */
-    @Deprecated // ... and optimized implementations already removed
+    default SchemaTableName getTableName(ConnectorSession session, ConnectorTableHandle table)
+    {
+        return getSchemaTableName(session, table);
```

Comment:
Why is the new method calling a deprecated method?

Reply:
because `getSchemaTableName` already existed and can be implemented by a connector

this PR is effectively rename (from ConnectorMetadata perspective) and this is how you handle a rename in a backwards-compatible way